camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/DslComponent.java

```diff
@@ -539,8 +550,9 @@ protected String resolveKeyName(boolean immediately) {
         @Override
         public final Maybe<Object> getImmediately() {
             Maybe<Entity> targetEntityMaybe = component.getImmediately();
-            if (targetEntityMaybe.isAbsent()) return Maybe.absent("Target entity not available");
+            if (targetEntityMaybe.isAbsent()) return Maybe.<Object>cast(targetEntityMaybe);
             EntityInternal targetEntity = (EntityInternal) targetEntityMaybe.get();
+            checkAndTagForRecursiveReference(targetEntity);
```

Comment:
the tag is left on the task.  you're right that could cause problems if the calling code isn't in a task (maybe it always will be but safer not to assume).  TODO remove tag afterwards